client/my-sites/pages/page-list.jsx

```diff
@@ -202,7 +213,7 @@ var Pages = React.createClass( {
 			rows = [];
 
 		// pages have loaded, sites have loaded, and we have a site instance or are viewing all-sites
-		if ( pages.length && this.props.sites.initialized ) {
+		if ( pages.length && this.props.hasSites ) {
```

Comment:
@mcsf I'm running into an issue here whereby I'm not able to accurately determine if I'm in the context of a single site or "all sites".

`state.ui.selectedSiteId === null` until the `siteSelection` controller is [pretty well along](https://github.com/Automattic/wp-calypso/blob/84c277ea4e3f2be4f58af65d2c99808cddcfccd1/client/my-sites/controller.js#L193).

Consequently, when I clear localStorage & load the pages section **WITH** a site slug in the URL, I'm seeing the results for "all sites" until that `SELECTED_SITE_SET` action gets dispatched.

Have you solve this elsewhere already? What am I missing?